src/coreclr/jit/fgbasic.cpp

```diff
@@ -684,24 +683,48 @@ void Compiler::fgReplaceJumpTarget(BasicBlock* block, BasicBlock* oldTarget, Bas
                 {
                     assert(block->KindIs(BBJ_COND));
                     block->SetTrueEdge(newEdge);
-
-                    if (oldEdge->hasLikelihood())
-                    {
-                        newEdge->setLikelihood(oldEdge->getLikelihood());
-                    }
                 }
             }
             else
             {
                 assert(block->FalseTargetIs(oldTarget));
                 FlowEdge* const oldEdge = block->GetFalseEdge();
 
+                if (block->TrueEdgeIs(oldEdge))
+                {
+                    // Branch was degenerate
+                    //
+                    // fgRemoveRefPred returns nullptr for BBJ_COND blocks with two flow edges to target
+                    fgRemoveConditionalJump(block);
+                    assert(block->KindIs(BBJ_ALWAYS));
+                    assert(block->TargetIs(oldTarget));
+                }
+
                 // fgRemoveRefPred should have removed the flow edge
                 fgRemoveRefPred(oldEdge);
                 assert(oldEdge->getDupCount() == 0);
                 FlowEdge* const newEdge = fgAddRefPred(newTarget, block, oldEdge);
-                block->SetFalseEdge(newEdge);
+
+                if (block->KindIs(BBJ_ALWAYS))
+                {
+                    block->SetTargetEdge(newEdge);
+                }
+                else
+                {
+                    assert(block->KindIs(BBJ_COND));
+                    block->SetFalseEdge(newEdge);
+                }
             }
+
+            if (block->GetFalseEdge() == block->GetTrueEdge())
```

Comment:
I think we need to check if `block` is still a `BBJ_COND` here before calling `GetFalseEdge/GetTrueEdge`. If we converted it to a `BBJ_ALWAYS` above, this will assert.

Reply:
Added the check.